### PR TITLE
Adds sign message to SDK demo

### PR DESCRIPTION
**Pull request by obber**
As title

### CODE REVIEW

File: components/sign-message.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const [decodedText, setDecodedText] = useState('');
+  const [encodedSignatureForUi, setEncodedSignatureForUi] =
+    useState<Uint8Array | null>();
+  const [publicKeyStr, setPublicKeyStr] = useState('');
+  const [verified, setVerified] = useState(false);
+  const [showDialog, setShowDialog] = useState(false);
+
+  const handleButtonClick = async () => {
+    const claim = textareaRef.current?.value ?? '';
+    const encodedText = new Uint8Array(Buffer.from(claim, 'utf-8'));
```
Comment (dborstelmann) [inline components/sign-message.tsx:25]: Is this typed?

**Review by dborstelmann — APPROVED** [review]

File: components/sign-message.tsx (hunk 1 above)
Comment (obber) [reply components/sign-message.tsx:25]: I'm unclear on what you're asking is typed.

Do you mean is the input into `signMessage` typed? If so, yes - it's typed as a Uint8Array.

Comment (dborstelmann) [reply components/sign-message.tsx:25]: Sorry I meant the string "utf-8".  I wish GitHub had character level highlighting rather than line.  Oh critique

Comment (obber) [reply components/sign-message.tsx:25]: Oh I see - yeah, the web API type definitions (ships with TypeScript) has the union type of string literals defined, and 'utf-8' is one of them.

Comment (obber) [reply components/sign-message.tsx:25]: i.e. when using `Buffer.from(..., secondArg)`, `secondArg` is a known list of string literals.

**Pull request merged**